### PR TITLE
get province list controller and env properties

### PR DESCRIPTION
**Pull request by sivakaruna**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Created rest property file and get province list controller and service

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Unit test

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **put the jira ticket # here**

### CODE REVIEW

File: src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.PropertySource;
 
 @SpringBootApplication
+@PropertySource("ecrc-rest.properties")
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java]: Siva. Can we change the name of the file to simply application.properties. There's no need to follow the former convention. I'd like to leave all properties in the same file.  Thanks.

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServicesImpl.java [hunk 2]
```diff
@@ -1,22 +1,56 @@
...
-}
 
+	public String getProvinceList() throws EcrcServiceException {
+		Mono<GetProvinceList> responseBody = this.webClient.get().uri(provinceListUri).retrieve()
+				.bodyToMono(GetProvinceList.class);
+		String response;
+		try {
+			response = objectMapper.writeValueAsString(responseBody.block());
+		} catch (JsonProcessingException e) {
+			throw new EcrcServiceException("JsonProcessingException");
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServicesImpl.java]: Add a //TODO here and a message indicating logging should be inserted above the stack trace please. Also, use the custom exception which allows for a message and a throwable(e), so we catch the entire error and why it occurred.

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/controller/GetProvinceListController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@RestController
+public class GetProvinceListController {
+	
+	@Autowired 
+	EcrcServices ecrcServices;
+
+	@CrossOrigin(origins = "/**")
+	@GetMapping("/getProvinceList")
+	public String getProvinceList() throws EcrcServiceException {
+		return ecrcServices.getProvinceList();
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/controller/GetProvinceListController.java]: I would wrap this in a try/catch and then find out what should be returned when things go bad from the service layer by checking into the original webMethods service. If you need help come get me.

File: src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java (hunk 1 above)
Comment (sivakaruna) [reply src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java]: Removed the new file

**Pull request merged**